lib/3scale/backend/service.rb

```diff
@@ -56,6 +56,14 @@ def load_by_id(service_id)
 
           return if service_attrs[:provider_key].nil?
 
+          # If service_attrs is nil then it means no entry has been found
+          # in the storage. In order to not break compatibility with existing
+          # Services, by default if no entry is found in the storage for the
+          # state attribute then the :active state is assumed
+          if service_attrs[:state].nil?
+            service_attrs[:state] = :active
+          end
```

Comment:
`service_attrs[:state] ||= :active`

Reply:
why don't you move this to the `initialize` method?
```ruby
def initialize(attributes = {})
  attributes[:state] ||= :active
  @state = attributes[:state]
  super attributes
end
```

Reply:
I did not move that to initialize because we want to set the state to suspended if the attribute does not exist (has not been passed as an attributes key) when the initialize method is directly called. The attributes[:state] ||= :active will set it to active it does not exist.

Reply:
that's why I added the two additional comments :) - let's have it sorted out in a single place, please

Reply:
I assumed that we wanted to differentiate this two cases with different behaviour:
* The case where the state entry does not exists in the database for an existing Service
* The case where a new Service is created via the initialize method and a state with nil value is intentionally set.

As we discussed, it is not possible to know if the second previous point is due to this has been done via initialize or because the service has been loaded via the load_by_id method and Redis simply has returned a nil value (meaning that Redis state key for that Service exists). Because of that we have decided to interpret a nil state or a not existing state as 'active'.

I have made the corresponding changes to achieve this and updated the tests.

Reply:
I have also updated the initial comment of this PR listing the possible situations that can happen